pkg/util/validation/limits.go

```diff
@@ -129,7 +130,13 @@ type Limits struct {
 	MetricRelabelConfigs                        []*relabel.Config   `yaml:"metric_relabel_configs,omitempty" json:"metric_relabel_configs,omitempty" doc:"nocli|description=List of metric relabel configurations. Note that in most situations, it is more effective to use metrics relabeling directly in the Prometheus server, e.g. remote_write.write_relabel_configs. Labels available during the relabeling phase and cleaned afterwards: __meta_tenant_id" category:"experimental"`
 	MetricRelabelingEnabled                     bool                `yaml:"metric_relabeling_enabled" json:"metric_relabeling_enabled" category:"experimental"`
 	ServiceOverloadStatusCodeOnRateLimitEnabled bool                `yaml:"service_overload_status_code_on_rate_limit_enabled" json:"service_overload_status_code_on_rate_limit_enabled" category:"experimental"`
-	IngestionArtificialDelay                    model.Duration      `yaml:"ingestion_artificial_delay" json:"ingestion_artificial_delay" category:"experimental" doc:"hidden"`
+
+	IngestionArtificialDelay                                         model.Duration `yaml:"ingestion_artificial_delay" json:"ingestion_artificial_delay" category:"experimental" doc:"hidden"`
+	IngestionArtificialDelayConditionForTenantsWithLessThanMaxSeries int            `yaml:"ingestion_artificial_delay_condition_for_tenants_with_less_than_max_series" json:"ingestion_artificial_delay_condition_for_tenants_with_less_than_max_series" category:"experimental" doc:"hidden"`
+	IngestionArtificialDelayDurationForTenantsWithLessThanMaxSeries  model.Duration `yaml:"ingestion_artificial_delay_duration_for_tenants_with_less_than_max_series" json:"ingestion_artificial_delay_duration_for_tenants_with_less_than_max_series" category:"experimental" doc:"hidden"`
+	IngestionArtificialDelayConditionForTenantsWithIDGreaterThan     int            `yaml:"ingestion_artificial_delay_condition_for_tenants_with_id_greater_than" json:"ingestion_artificial_delay_condition_for_tenants_with_id_greater_than" category:"experimental" doc:"hidden"`
+	IngestionArtificialDelayDurationForTenantsWithIDGreaterThan      model.Duration `yaml:"ingestion_artificial_delay_duration_for_tenants_with_id_greater_than" json:"ingestion_artificial_delay_duration_for_tenants_with_id_greater_than" category:"experimental" doc:"hidden"`
```

Comment:
I think this option is potentially _very_ dangerous as it becomes extremely easy to be greedy with the selection - I was going to suggest logging for this condition so that we can audit the ID selection but this has the potential to be extremely expensive on such a hot path.

I'll leave the judgement up to you.

Reply:
I don't see it more risky than the global config option that applies the same latency to **all** tenants. This option, at most, will apply it to all tenants. What risk do you foresee? Any practical example?

Reply:
When you set it globally, it's because you _really_ want to set it for all tenants. 

Maybe, I don't understand the practical use case well enough - but I'm thinking about a scenario where I want to select IDs `1`, `2` and `3` so I set it to `0` and then either `4` gets created in gcom or I don't realise there was a `4` to begin with so that also ends up with the latency when I originally didn't want to select it. 

It might as well be safe to select an ID that was newly created or wasn't intended, but the point stand that you select something you weren't expecting.

As I said, I don't have a problem with it and feel free to merge/continue it was mostly an observation.

Reply:
Ok let's discuss it offline.

Reply:
Discussed offline. Josh is fine.